libs/providers/openstack.py

```
@@ -269,29 +342,79 @@ def clone_vm(
             LOGGER.info(f"Creating snapshot '{snapshot_name}'...")
             snapshot = self.api.compute.create_server_image(server=source_vm.id, name=snapshot_name, wait=True)
 
-            if not snapshot:
-                raise ValueError(f"Could not create snapshot '{snapshot_name}' for VM '{source_vm_name}'.")
+            # Get all volume snapshots created by create_server_image()
+            volume_snapshot_name = f"snapshot for {snapshot_name}"
+            volume_snapshots = list(self.api.block_storage.snapshots(name=volume_snapshot_name))
+
+            if not volume_snapshots:
+                raise ValueError(f"No volume snapshots found for '{volume_snapshot_name}'.")
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)